plugins/item_tasks/web_client/views/ItemSelectorWidget.js

```diff
@@ -0,0 +1,148 @@
+import _ from 'underscore';
+
+import { getCurrentUser } from 'girder/auth';
+import BrowserWidget from 'girder/views/widgets/BrowserWidget';
+import View from 'girder/views/View';
+import ItemModel from 'girder/models/ItemModel';
+import { restRequest } from 'girder/rest';
+
+var ItemSelectorWidget = View.extend({
+    events: {
+        'submit .g-new-file-select-form': '_selectButton'
+    },
+
+    initialize: function (settings) {
+        if (!this.model) {
+            this.model = new ItemModel();
+        }
+    },
+
+    render: function () {
+        var showItems = false,
+            input = false,
+            preview = true,
+            type = this.model.get('type'),
+            title, help;
+
+        // Customize the browser widget according the argument type
+        if (type === 'item' || type === 'file' || type === 'image') {
+            showItems = true;
+            title = 'Select an item';
+            help = 'Click on an item to select it, then click "Save"';
+        }
+
+        if (type === 'directory') {
+            title = 'Select a folder';
+            help = 'Browse to a directory to select it, then click "Save"';
+        }
+
+        if (type === 'new-file') {
+            title = 'Create a new item';
+            help = 'Browse to a path, enter a name, then click "Save"';
+            input = {
+                label: 'Name',
+                placeholder: 'Choose a name for the new item',
+                default: this.model.get('fileName')
+            };
+            preview = false;
+        }
+
+        this._browserWidget = new BrowserWidget({
+            el: $('#g-dialog-container'),
+            parentView: this,
+            showItems: showItems,
+            selectItem: showItems,
+            root: getCurrentUser(),
+            titleText: title,
+            helpText: help,
+            input: input,
+            showPreview: preview,
+            validate: _.bind(this._validateSelection, this)
+        }).render();
+        this._browserWidget.once('g:saved', (model, inputValue) => {
+            this._browserWidget.$el.modal('hide');
+            this.model.set({
+                value: model,
+                fileName: inputValue || model.name()
+            });
+            this.trigger('g:saved');
+        });
+
+        return this;
+    },
+
+    _validateSelection: function (item) {
+        switch (this.model.get('type')) {
+            case 'file':
+                return restRequest({path: '/item/' + item.id + '/files', data: {limit: 1}})
```

Comment:
In light of https://github.com/girder/girder/pull/2273 , can you make this parameter be `url: ` instead of `path: `?

Can you also indent the parameters, so each is on a separate line?